### PR TITLE
Fix Start-AzAutomationDscCompilationJob (Issue #8347)

### PR DESCRIPTION
**Pull request by lwajswaj**
## Description

This PR fixed issue #8347; the root cause of this issue is a synchronous call to 'create' method which, when job gets suspended, it never releases control as the job hasn't completed. Not sure if this is by-design on the SDK or it's a bug. Using 'CreateAsync' fixes it but you cannot do a regular Wait() or GetWaiter() as the same behavior is repeated. Base on the documentation [Compiling a DSC Configuration with Windows PowerShell](https://docs.microsoft.com/en-us/azure/automation/automation-dsc-compile#compiling-a-dsc-configuration-with-windows-powershell), the original idea of this cmdlet seemed to be just starting the job but not waiting till it completed as it was the previous behavior. This PR changes Start-AzAutomationDscCompilationJob to just submit the compilation job but not waiting for its completion to return control, after this change this is how the usage looks like:

```powershell
PS C:\Program Files\WindowsPowershellCore> Import-Module C:\Users\Leandro\Source\Repos\GitHub\lwajswaj\azure-powershell\artifacts\Debug\Az.Automation\Az.Automation.psd1
PS C:\Program Files\WindowsPowershellCore> Start-AzAutomationDscCompilationJob -ConfigurationName GoodConfig -ResourceGroupName <ResourceGroupName> -AutomationAccountName <AutomationAccountName>


ResourceGroupName      : <ResourceGroupName>
AutomationAccountName  : <AutomationAccountName>
Id                     : f6e2a35b-b076-4c98-b211-5975d5fa177e
CreationTime           : 22/02/2019 21:41:24 -03:00
Status                 : New
StatusDetails          : None
StartTime              :
EndTime                :
Exception              :
LastModifiedTime       : 22/02/2019 21:41:24 -03:00
LastStatusModifiedTime : 23/02/2019 00:41:24 +00:00
JobParameters          : {}
ConfigurationName      : GoodConfig



PS C:\Program Files\WindowsPowershellCore> Start-AzAutomationDscCompilationJob -ConfigurationName BadConfig -ResourceGroupName <ResourceGroupName> -AutomationAccountName <AutomationAccountName>


ResourceGroupName      : <ResourceGroupName>
AutomationAccountName  : <AutomationAccountName>
Id                     : f16827d2-3eab-4e1c-bbfc-673c6e4bf14c
CreationTime           : 22/02/2019 21:41:34 -03:00
Status                 : New
StatusDetails          : None
StartTime              :
EndTime                :
Exception              :
LastModifiedTime       : 22/02/2019 21:41:34 -03:00
LastStatusModifiedTime : 23/02/2019 00:41:34 +00:00
JobParameters          : {}
ConfigurationName      : BadConfig



PS C:\Program Files\WindowsPowershellCore> Get-AzAutomationDscCompilationJob -Id f16827d2-3eab-4e1c-bbfc-673c6e4bf14c -ResourceGroupName <ResourceGroupName> -AutomationAccountName <AutomationAccountName>


ResourceGroupName      : <ResourceGroupName>
AutomationAccountName  : <AutomationAccountName>
Id                     : f16827d2-3eab-4e1c-bbfc-673c6e4bf14c
CreationTime           : 22/02/2019 21:41:34 -03:00
Status                 : Suspended
StatusDetails          : None
StartTime              : 22/02/2019 21:42:11 -03:00
EndTime                :
Exception              : Exception calling "NewScriptBlock" with "1" argument(s): "At line:7 char:5
                         +     Import-DscResource -ModuleName 'PSDscResources12345'
                         +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                         Could not find the module 'PSDscResources12345'." (At line:7 char:5
                         +     Import-DscResource -ModuleName 'PSDscResources12345'
                         +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                         Could not find the module 'PSDscResources12345'.)
LastModifiedTime       : 22/02/2019 21:42:14 -03:00
LastStatusModifiedTime : 23/02/2019 00:42:14 +00:00
JobParameters          : {}
ConfigurationName      : BadConfig



PS C:\Program Files\WindowsPowershellCore> Get-AzAutomationDscCompilationJob -Id f6e2a35b-b076-4c98-b211-5975d5fa177e -ResourceGroupName <ResourceGroupName> -AutomationAccountName <AutomationAccountName>


ResourceGroupName      : <ResourceGroupName>
AutomationAccountName  : <AutomationAccountName>
Id                     : f6e2a35b-b076-4c98-b211-5975d5fa177e
CreationTime           : 22/02/2019 21:41:24 -03:00
Status                 : Completed
StatusDetails          : None
StartTime              : 22/02/2019 21:42:06 -03:00
EndTime                : 22/02/2019 21:42:15 -03:00
Exception              :
LastModifiedTime       : 22/02/2019 21:42:15 -03:00
LastStatusModifiedTime : 23/02/2019 00:42:15 +00:00
JobParameters          : {}
ConfigurationName      : GoodConfig
```

This PR introduces a behavior change in the cmdlet for which a breaking change warning has been included.
## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
@lwajswaj Hey Leandro, thanks for the contribution!

@avkaur @vrdmr would one of you be able to take a look at these changes?

**Comment by lwajswaj** [comment]
Any update on this PR review?

**Comment by cormacpayne** [comment]
@lwajswaj the proper contacts from the Automation team have been contacted offline -- @avkaur @mgreenegit @nitinbps

**Comment by mgreenegit** [comment]
Just FYI, I opened a work item to track this for our team to make sure we get this reviewed.

**Comment by vladimir-shcherbakov** [comment]
@lwajswaj What is the status of the PR? Are you still waiting for the @vrdmr review?

**Comment by lwajswaj** [comment]
@vladimir-shcherbakov, 2 out of the 4 reviewers approved it. However, I don't have rights to merge it as I'm not an authorized user.

File: src/Automation/Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs [hunk 1]
```diff
@@ -25,6 +26,7 @@ namespace Microsoft.Azure.Commands.Automation.Cmdlet
     /// <summary>
     /// starts azure automation compilation job
     /// </summary>
+    [GenericBreakingChange("Behavior changed", ChangeDescription = "Cmdlet returns control to pipeline before compilation job has completed. Refer to https://docs.microsoft.com/en-us/azure/automation/automation-dsc-compile#compiling-a-dsc-configuration-with-windows-powershell")]
     [Cmdlet("Start", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "AutomationDscCompilationJob", SupportsShouldProcess = true)]
```
Comment (vladimir-shcherbakov) [inline src/Automation/Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs]: This line confuses me - if it's a breaking change we can't merge into master.

Comment (lwajswaj) [reply src/Automation/Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs]: Check the issue this PR solves. Apparently, according to [documentation](https://docs.microsoft.com/en-us/azure/automation/automation-dsc-compile#compiling-a-dsc-configuration-with-windows-powershell), the intent of the cmdlet was just to start the job but not wait for its completion. However, the actual implementation waits for completion which, when the compilation job get suspended, it never ends, causing the need to just close the PS window. This change returns the cmdlet to its original intent.

Comment (lwajswaj) [reply src/Automation/Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs]: With current implementation, the documented wait-loop is useless

Comment (vladimir-shcherbakov) [reply src/Automation/Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs]: @lwajswaj  if it's not a breaking change - please remove the breaking change attribute.

Comment (lwajswaj) [reply src/Automation/Automation/Cmdlet/StartAzureAutomationDscCompilationJob.cs]: @vladimir-shcherbakov , done. I've just removed the attribute

**Comment by vladimir-shcherbakov** [comment]
@lwajswaj Is it possible for you to cover basic scenario with tests?

**Comment by vladimir-shcherbakov** [comment]
@lwajswaj  Ping on the comments above.

**Comment by MiYanni** [comment]
@lwajswaj No activity for 2 weeks. Closing. Feel free to reopen when you have time to address the comments above.

**Comment by lwajswaj** [comment]
Hello, I've been with some business trips + vacations and it was not possible to stay on top of this PR. How should I proceed? opening a new one or can this one be re-open?

**Pull request closed without merging**